perception/autoware_tensorrt_rtmdet/include/tensorrt_rtmdet/tensorrt_rtmdet.hpp

```diff
@@ -0,0 +1,183 @@
+
+  bool doInferenceWithRoi(
+    const std::vector<cv::Mat> & images, ObjectArrays & objects, cv::Mat & mask,
+    std::vector<uint8_t> & class_ids, const std::vector<cv::Rect> & roi);
+
+  bool doMultiScaleInference(
+    const cv::Mat & image, ObjectArrays & objects, const std::vector<cv::Rect> & roi);
+
+  inline ColorMap getColorMap() { return color_map_; }
+
+  void getColorizedMask(const ColorMap & color_map, const cv::Mat & mask, cv::Mat & color_mask);
+
+  void printProfiling(void);
+
+private:
+  void preprocess(const std::vector<cv::Mat> & images);
+
+  void preprocessGpu(const std::vector<cv::Mat> & images);
+
+  void preprocessWithRoi(const std::vector<cv::Mat> & images, const std::vector<cv::Rect> & rois);
+
+  void preprocessWithRoiGpu(
+    const std::vector<cv::Mat> & images, const std::vector<cv::Rect> & rois);
+
+  void multiScalePreprocess(const cv::Mat & image, const std::vector<cv::Rect> & rois);
+
+  void multiScalePreprocessGpu(const cv::Mat & image, const std::vector<cv::Rect> & rois);
+
+  bool multiScaleFeedforward(const cv::Mat & image, int batch_size, ObjectArrays & objects);
+
+  bool feedforward(
+    const std::vector<cv::Mat> & images, ObjectArrays & objects, cv::Mat & mask,
+    std::vector<uint8_t> & class_ids);
+
+  void readColorMap(const std::string & color_map_path);
+
+  void nmsSortedBboxes(const ObjectArray & input_objects, ObjectArray & output_objects) const;
+
+  std::unique_ptr<tensorrt_common::TrtCommon> trt_common_;
+
+  std::vector<float> input_h_;
+  CudaUniquePtr<float[]> input_d_;
+  CudaUniquePtr<float[]> out_dets_d_;
+  CudaUniquePtr<int32_t[]> out_labels_d_;
+  CudaUniquePtr<float[]> out_masks_d_;
+
+  size_t out_elem_num_;
+  size_t out_elem_num_per_batch_;
+  CudaUniquePtr<float[]> out_prob_d_;
+
+  StreamUniquePtr stream_{makeCudaStream()};
+
+  int32_t max_detections_;
+  float scale_width_;
+  float scale_height_;
+
+  // size of input image for model
+  int model_input_width_;
+  int model_input_height_;
+
+  int num_class_;
+  float score_threshold_;
+  float nms_threshold_;
+  float mask_threshold_;
+  int batch_size_;
+  CudaUniquePtrHost<float[]> out_prob_h_;
+
+  // flag whether preprocess are performed on GPU
+  bool use_gpu_preprocess_;
+  // host buffer for preprocessing on GPU
+  CudaUniquePtrHost<unsigned char[]> image_buf_h_;
+  // device buffer for preprocessing on GPU
+  CudaUniquePtr<unsigned char[]> image_buf_d_;
+  // normalization factor used for preprocessing
+  double norm_factor_;
+
+  std::vector<int> output_strides_;
+
+  int src_width_;
+  int src_height_;
+
+  std::vector<float> mean_;
+  std::vector<float> std_;
+
+  // host pointer for ROI
+  CudaUniquePtrHost<Roi[]> roi_h_;
+  // device pointer for ROI
+  CudaUniquePtr<Roi[]> roi_d_;
+
+  // Host pointer for outputs
+  std::unique_ptr<float[]> out_dets_h_;
+  std::unique_ptr<int32_t[]> out_labels_h_;
+  std::unique_ptr<float[]> out_masks_h_;
+
+  // Segmentation
+  ColorMap color_map_;
```

Comment:
Thank you for your PR!

Please add `const` to all variables whose values will not change after initialization.

Reply:
`const` has been used in places where it is applicable.

related commit: https://github.com/autowarefoundation/autoware.universe/pull/8165/commits/42fd33e9814fe63a9ae1f5cbdec85bd5f5d85fd0